activerecord/lib/active_record/connection_adapters/postgresql/oid/timestamp.rb

```diff
@@ -8,6 +8,13 @@ class Timestamp < DateTime # :nodoc:
           def type
             real_type_unless_aliased(:timestamp)
           end
+
+          def ==(other)
```

Comment:
Would not this only fix the problem with the postgresql timestamp? I think the problem happens with any type that uses `TimeZoneConverter`, which also includes `DateTime`

Reply:
Got it, @rafaelfranca 

I'll push a new fix for reviews quite soon